File: Easy/0455-Assign Cookies/Greedy.rs
impl Solution {
    pub fn find_content_children(g: Vec<i32>, s: Vec<i32>) -> i32 {
        let mut g = g;
        let mut s = s;
        g.sort_unstable();
        s.sort_unstable();

        let mut i = 0;
        let mut j = 0;
        let mut ret = 0;

        while i < g.len() && j < s.len() {
            if s[j] >= g[i] {
                ret += 1;
                i += 1;
            }
            j += 1;
        }

        ret
    }
}

File: Easy/0455-Assign Cookies/Greedy.rb
# @param {Integer[]} g
# @param {Integer[]} s
# @return {Integer}
def find_content_children(g, s)
    i, j = 0, 0
    ret = 0

    g.sort!
    s.sort!

    while i < g.length and j < s.length
        if s[j] >= g[i]
            ret += 1
            i += 1
        end
        j += 1
    end

    return ret
end
